File: apps/backend/src/EnterpriseAgentOs.Domain/Interfaces/AgentTemplates/IAgentTemplateService.cs
namespace EnterpriseAgentOs.Domain.Interfaces.AgentTemplates;

public interface IAgentTemplateService
{
    Task<IReadOnlyList<AgentTemplateDto>> ListAsync(CancellationToken ct = default);
    Task<AgentTemplateDto?> GetAsync(Guid id, CancellationToken ct = default);
    Task<EnterpriseAgentOs.Domain.DTOs.Agents.AgentDto> CreateAgentFromTemplateAsync(
        Guid templateId,
        string name,
        string provider,
        string? model,
        Guid ownerId,
        CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentTemplateService.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.dtos.AgentTemplateDto;

public interface IAgentTemplateService {
    CompletableFuture<List<AgentTemplateDto>> listAsync();
    CompletableFuture<AgentTemplateDto> getAsync(UUID id);
}
